internal/handlers/common/params.go

```diff
@@ -93,3 +93,95 @@ func GetNumberParam(parameterName string, value any) (int64, error) {
 	}
 	return numberValue, nil
 }
+
+// getBinaryMaskParam matches value type, returning types.Binary and error if match failed.
+// Possible values are: position array ([1,3,5] == 010101), whole number value and types.Binary value.
+func getBinaryMaskParam(value any) (types.Binary, error) {
+	var mask types.Binary
+	var err error
+
+	switch value := value.(type) {
+	case *types.Array:
+		// {field: {$bitsAllClear: [position1, position2]}}
+		mask, err = types.BinaryFromArray(value)
+		if err != nil {
+			return types.Binary{}, NewError(ErrBadValue, err)
+		}
+
+	case int32:
+		// {field: {$bitsAllClear: bitmask}}
+		if value < 0 {
+			return types.Binary{}, NewErrorMsg(ErrFailedToParse,
+				fmt.Sprintf(`Expected a positive number in: $bitsAllClear: %#v`, value),
+			)
+		}
+
+		mask, err = types.BinaryFromInt(int64(value))
+		if err != nil {
+			return types.Binary{}, NewError(ErrBadValue, err)
+		}
+	case int64:
+		// {field: {$bitsAllClear: bitmask}}
+		if value < 0 {
+			return types.Binary{}, NewErrorMsg(ErrFailedToParse,
+				fmt.Sprintf(`Expected a positive number in: $bitsAllClear: %#v`, value),
+			)
+		}
+
+		mask, err = types.BinaryFromInt(value)
+		if err != nil {
+			return types.Binary{}, NewError(ErrBadValue, err)
+		}
+	case float64:
+		// TODO check float negative zero
+		if value != math.Trunc(value) || math.IsNaN(value) || math.IsInf(value, 0) {
+			return types.Binary{}, NewErrorMsg(ErrFailedToParse,
+				fmt.Sprintf("Expected an integer: $bitsAllClear: %#v", value),
+			)
+		}
+		mask, err = types.BinaryFromInt(int64(value))
+		if err != nil {
+			return types.Binary{}, err
+		}
+	case types.Binary:
+		// {field: {$bitsAllClear: BinData()}}
+		mask = value
+	default:
+		return types.Binary{}, NewErrorMsg(ErrBadValue,
+			fmt.Sprintf(
+				`value takes an Array, a number, or a BinData but received: $bitsAllClear: %#v`, value),
+		)
+	}
+	return mask, nil
+}
+
+// GetBinaryParam matches value type, returning types.Binary and error if match failed.
+func GetBinaryParam(value any) (types.Binary, error) {
```

Comment:
Seems unused

Reply:
It's used in `filterFieldExprBitsAllClear`

Reply:
Please not resolve the conversation if you want your answer to be seen :)